ably/types/options.py

```
@@ -9,15 +9,20 @@
+                 tls_port=0, use_binary_protocol=True, queue_messages=False, recover=False, endpoint=None,
+                 environment=None, http_open_timeout=None, http_request_timeout=None,
+                 realtime_request_timeout=None, http_max_retry_count=None, http_max_retry_duration=None,
+                 fallback_hosts=None, fallback_retry_timeout=None, disconnected_retry_timeout=None,
+                 idempotent_rest_publishing=None, loop=None, auto_connect=True,
+                 suspended_retry_timeout=None, connectivity_check_url=None,
                  channel_retry_timeout=Defaults.channel_retry_timeout, add_request_ids=False, **kwargs):
 
         super().__init__(**kwargs)
 
+        if endpoint is not None:
+            if environment is not None or rest_host is not None or realtime_host is not None:
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)